wallet/wallet.go

```diff
@@ -2359,6 +2359,149 @@ func (w *Wallet) GetTransactions(startBlock, endBlock *BlockIdentifier,
 	return &res, err
 }
 
+// GetTransaction returns data if any transaction given its id. A mined
+// transaction is returned in a Block structure which records properties about
+// the block. A bool is also returned to denote if the transaction did exist
+// in the database or not.
+func (w *Wallet) GetTransaction(txHash *chainhash.Hash) (*GetTransactionsResult,
+	bool, error) {
+
+	var res GetTransactionsResult
+	chainClient := w.chainClient
+	if chainClient == nil {
+		return nil, false, errors.New("no chain server client")
+	}
+
+	// Get the transaction information from directly calling the backend
+	// endpoint.
+	var txResult *btcjson.TxRawResult
+	var err error
+	switch client := chainClient.(type) {
+	case *chain.RPCClient:
+		txResult, err = client.GetRawTransactionVerbose(txHash)
+		if err != nil {
+			return nil, false, err
+		}
+	case *chain.BitcoindClient:
+		txResult, err = client.GetRawTransactionVerbose(txHash)
+		if err != nil {
+			return nil, false, err
+		}
+	case *chain.NeutrinoClient:
+		return nil, false, errors.New("not supported with neutrino client")
+	}
+
+	// Set the block hash from the provided string and get the block height
+	// from it.
+	var blockHash *chainhash.Hash
+	var blockHeight int32 = -1
+	if txResult.BlockHash != "" {
+		blockHashBytes, err := hex.DecodeString(txResult.BlockHash)
+		if err != nil {
+			return nil, false, err
+		}
+
+		// The byte is reversed due to different endianness in response
+		// and call.
+		reversed := make([]byte, len(blockHashBytes))
+		for i, n := range blockHashBytes {
+			j := len(blockHashBytes) - i - 1
+			reversed[j] = n
+		}
+		blockHash, err = chainhash.NewHash(reversed)
+		if err != nil {
+			return nil, false, err
+		}
+
+		// Obtain the block height from the backend.
+		switch client := chainClient.(type) {
+		case *chain.RPCClient:
+			header, err := client.GetBlockHeaderVerbose(blockHash)
+			if err != nil {
+				return nil, false, err
+			}
+			blockHeight = header.Height
+		case *chain.BitcoindClient:
+			var err error
+			blockHeight, err = client.GetBlockHeight(blockHash)
```

Comment:
It actually depends on if a return variable would otherwise overshadow a variable in the scope or not. In this case we are interested in the block height, which is directly returned for the `BitcoindClient` and therefore directly set to the variable `blockHeight`. But for the `RPCClient` instead we first have to declare a local return variable `header` with `:=` that contains the height and then set the variable `blockHeight` in the scope to its `Height` value.